src/client.ts

```diff
@@ -3787,18 +3788,18 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
 
         // If we expect that an event is part of a thread but is missing the relation
         // we need to add it manually, as well as the reply fallback
-        if (threadId && !content["m.relates_to"]?.rel_type) {
-            const isReply = !!content["m.relates_to"]?.["m.in_reply_to"];
-            content["m.relates_to"] = {
-                ...content["m.relates_to"],
+        if (threadId && !content!["m.relates_to"]?.rel_type) {
+            const isReply = !!content!["m.relates_to"]?.["m.in_reply_to"];
+            content!["m.relates_to"] = {
```

Comment:
If we're making `content` optional, what's supposed to happen in this case?

Reply:
`content` is only optional in the final signature, in all the callable overloads it must be specified, therefore it isn't really optional but overloads are strange in typescript

Reply:
It might be cleaner if the arg changes name to `contentOrTxnId` and then we create a better-typed `content` variable at the top of the method but then the param names get messy in the overload itself